packages/smooth_app/lib/cards/product_cards/knowledge_panels/knowledge_panel_title_card.dart

```diff
@@ -17,15 +17,20 @@ class KnowledgePanelTitleCard extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final ThemeData themeData = Theme.of(context);
+    final ColorScheme colorScheme = themeData.colorScheme;
     Color? colorFromEvaluation;
     if (evaluation != null &&
         (knowledgePanelTitleElement.iconColorFromEvaluation ?? false)) {
-      colorFromEvaluation = _getColorFromEvaluation(evaluation!);
+      if (themeData.brightness == Brightness.dark) {
+        colorFromEvaluation = _getColorFromEvaluationDarkMode(evaluation!);
+      } else {
+        colorFromEvaluation = _getColorFromEvaluation(evaluation!);
+      }
     }
     if (evaluation != null &&
         _getColorFromEvaluation(evaluation!) == null &&
         themeData.brightness == Brightness.dark) {
-      colorFromEvaluation = Colors.white;
+      colorFromEvaluation = colorScheme.onBackground;
```

Comment:
Please get rid of that. We get the color from either `_getColorFromEvaluation` or `_getColorFromEvaluationDarkMode`.

Reply:
Now color is retrieved from `_getColorFromEvaluationDarkMode`.